Automattic/Gravatar-SDK-Android
# Create docs.yml workflow to generate API documentation

Closes #371

### Description

This PR adds a `docs.yml` file that will be triggered on each "stable" tag push and will build the API documentation. 

This should be a fully automated process now so the release steps have been updated accordingly. 

I did test this workflow first on branch pushes and a different docs branch - you can the commits generated here https://github.com/Automattic/Gravatar-SDK-Android/commits/adam/only_docs/. Note that they have the wrong version in the commit due to some issues with `git describe` - but we don't use this here, it was only for testing. 

### Testing Steps

It's worth testing that the `sdkVersion` is properly set in the QE, so run it and inspect the network requests.

As for the workflow itself - not sure. I tested it as mentioned above, so now, I guess we will see once we create a new release :D
